### ISSUE TITLE
Navigation: wrong 180 degrees turn (Øysand, Norway)

### ISSUE THREAD

**Issue by gunnar22** [osmandapp/OsmAnd#9622]
See screenshot

![wrong-180-degrees-turn](https://user-images.githubusercontent.com/30504902/90070779-b4b34580-dcf4-11ea-9e22-a892401dcdde.jpg)

OsmAnd routes forward (west) to the main road and then a 180 degrees (impossible) turn there.
Correct routing would be to take the bridge (south), which crosses over the main road, and then the link (east) onto the main road (as I have drawn with a green line). 

Starting point coordinates: 63.32415, 10.21648
https://www.openstreetmap.org/#map=17/63.32415/10.21648
(I temporarily parked at the bus stop there to start the routing to turn around and drive eastwards.)
I used OsmAnd's in-app offline routing.
OsmAnd+ on android, the version and map (live update) as of date: 2020-07-18
Routing profile: car.

Maybe similar to #9578

**Comment by Hedaja** [comment]
There are missing u-turn-restrictions  and other turn restrictions for this crossing in the OSM data. OsmAnd probably routes you this way because doing one left turn is cheaper than two. 
You can add them fairly easily with the browser editor of OSM. Do you know how to do it and wanna give it a shot?
If not, I can take over and fix it.

**Comment by Hedaja** [comment]
There were a whole more turn restrictions missing all along E39.
I have added them now. This should fix the problem. But some insight in turn penalties for this case might still be useful

**Comment by gunnar22** [comment]
@Hedaja, Thanks for your kindness regarding editing, however I know how to do it. But instead of doing patchwork in the OSM map data, I am thinking bigger. It would be more robust to cover this through algorithm heuristics. Then it would work everywhere. I agree with @dead10ck: https://github.com/osmandapp/OsmAnd/issues/9235#issuecomment-670821494 . So, how about making it more 'expensive' in the algorithm to take one 180 deg. turn instead of two 90? And/or make it very expensive to take a 180 degree turn from a link onto the main road (a kind of driving that nobody would do, and probably most OSM mappers would not tag because it is so obvious and seems unnecessary), which should be possible to infer automatically from data without special tags. 

![image](https://user-images.githubusercontent.com/30504902/90319460-bc632c00-df38-11ea-8204-61c20fafa3d7.png)
No-left-turn will probably not be tagged at many places similar to this because it seems obvious and unnecessary. Instead let the algorithm understand that 180 deg. turn from link to road is expensive.

**Comment by vshcherb** [comment]
Yes, that's quite bad and it's similar to introduce separate penalty for sharp turns. 
Important note: it's not standard route cause it includes visual "U-Turn", so it looks that driver missed a correct turn and need to make a U-turn now

**Comment by gunnar22** [comment]
I don't understand vshcherb's comment. What was quite bad?
The driver did not miss a correct turn. See first post here.
